packages/payments-plugin/src/mollie/mollie.service.ts

```diff
@@ -291,6 +264,19 @@ export class MollieService {
                 `Unable to find order ${mollieOrder.orderNumber}, unable to process Mollie order ${mollieOrder.id}`,
             );
         }
+        if (order.orderPlacedAt) {
+            const paymentWithSameTransactionId = order.payments.find(
+                p => p.transactionId === mollieOrder.id && p.state === 'Settled',
+            );
+            if (!paymentWithSameTransactionId) {
+                // The order is paid for again, with another transaction ID. This means the customer paid twice
+                Logger.error(
+                    `Order '${order.code}' is already paid. Mollie order '${mollieOrder.id}' should be refunded.`,
+                    loggerCtx,
+                );
+                return;
+            }
```

Comment:
Is there a way we can tap into this error log and automatically report this to Sentry?

Reply:
I think the wrapper logger would work well, since I also need it for some other logs. Thanks for the suggestion!

Reply:
The log message is checked in the test. I will add a comment that the message shouldn't be changed because monitoring can depend on it.